nifi-toolkit/nifi-toolkit-admin/src/main/groovy/org/apache/nifi/toolkit/admin/client/NiFiClientUtil.groovy

```diff
@@ -0,0 +1,156 @@
+            urlBuilder.append(StringUtils.isEmpty(niFiProperties.getProperty(NiFiProperties.WEB_HTTPS_PORT)) ? "8080": niFiProperties.getProperty(NiFiProperties.WEB_HTTPS_PORT))
+        }
+
+        if(!StringUtils.isEmpty(endpoint)) {
+            urlBuilder.append(endpoint)
+        }
+
+        urlBuilder.toString()
+    }
+
+    public static String getUrl(NiFiProperties niFiProperties, NodeDTO nodeDTO, String endpoint){
+
+        final StringBuilder urlBuilder = new StringBuilder();
+        if(!StringUtils.isEmpty(niFiProperties.getProperty(NiFiProperties.WEB_HTTPS_PORT))){
+            urlBuilder.append("https://")
+
+        }else{
+            urlBuilder.append("http://")
+        }
+        urlBuilder.append(nodeDTO.address)
+        urlBuilder.append(":")
+        urlBuilder.append(nodeDTO.apiPort)
+
+        if(!StringUtils.isEmpty(endpoint)) {
+            urlBuilder.append(endpoint)
+        }
+
+        urlBuilder.toString()
+    }
+
+    public static ClusterEntity getCluster(final Client client, NiFiProperties niFiProperties, List<String> activeUrls){
+
+        if(activeUrls.isEmpty()){
+            final String url = getUrl(niFiProperties,null)
+            activeUrls.add(url)
+        }
+
+        for(String activeUrl: activeUrls) {
+
+            try {
+
+                String url = activeUrl + GET_CLUSTER_ENDPOINT
+                final WebResource webResource = client.resource(url)
+                final ClientResponse response = webResource.type("application/json").get(ClientResponse.class)
+
+                Integer status = response.getStatus()
+
+                if (status != 200) {
+                    if (status == 404) {
+                        logger.warn("This node is not attached to a cluster. Please connect to a node that is attached to the cluster for information")
+                    } else {
+                        logger.warn("Failed with HTTP error code: {}, message: {}", status, response.getStatusInfo().getReasonPhrase())
+                    }
+                } else if (status == 200) {
+                    return response.getEntity(ClusterEntity.class)
+                }
+
+            }catch(Exception ex){
+                logger.warn("Exception occurred during connection attempt: {}",ex.localizedMessage)
+            }
+
+        }
+
+        throw new RuntimeException("Unable to obtain cluster information")
+
+    }
+
+    public static List<String> getActiveClusterUrls(final Client client, NiFiProperties niFiProperties){
+
+        final ClusterEntity clusterEntity = getCluster(client, niFiProperties, Lists.newArrayList())
+        final List<NodeDTO> activeNodes = clusterEntity.cluster.nodes.findAll{ it.status == "CONNECTED" }
+        final List<String> activeUrls = Lists.newArrayList()
+
+        activeNodes.each {
+            activeUrls.add(getUrl(niFiProperties,it, null))
+        }
+        activeUrls
+    }
+
+    public static String convertToJson(NodeDTO nodeDTO){
```

Comment:
@brosander I originally had an issue with Jackson and date serialization however I figured out what I needed to do. It's now in use in the recent commit.